include/vector2d.h

```diff
@@ -414,5 +415,20 @@ class vector2d
 } // end namespace core
 } // end namespace irr
 
+namespace std
+{
+
+template<class T>
+struct hash<irr::core::vector2d<T> >
+{
+	size_t operator()(const irr::core::vector2d<T>& vec) const
+	{
+		size_t h1 = hash<T>()(vec.X);
+		size_t h2 = hash<T>()(vec.Y);
+		return (h1 << (4 * sizeof(h1)) | h1 >> (4 * sizeof(h1))) ^ h2;

```

Comment:
How good is this hash combination?
Have you thought about doing the same as boost:
https://github.com/boostorg/container_hash/blob/e9ffeceeab96ecd6979393676150874c3e5cd247/include/boost/container_hash/hash.hpp#L316

Reply:
I don't see any visible collisions, the code on the link does almost the same + some magic number.